### DISCUSSION TITLE
403

### DISCUSSION THREAD

**Discussion by benlabbeus** [shibayan/appservice-acmebot#299] [Q&A] [answered]
Getting a 403 when attempting to access /add-certificate.

Went through the readme verbatim. Deleted and retried, still 403.

Giving the Function "Owner" role still gives me a 403 when attempting to access.

Any help much appreciated!

**Comment by shibayan** [comment, marked as answer]
Authentication setup required
https://github.com/shibayan/appservice-acmebot#2-enable-app-service-authentication

**Comment by benlabbeus** [comment]
Thank you for the quick response!

Authentication \ Authorization on the Function has been enabled with "Express"
![image](https://user-images.githubusercontent.com/5489053/109993831-8bbec600-7cc1-11eb-8332-fe5116fae103.png)

I've also applied Website Contributor and Web Plan Contributor to the Function App in the IAM of the Resource Group. (Only using one Resource Group). I don't get the option for Subscription, but I only have one.
![image](https://user-images.githubusercontent.com/5489053/109994219-f5d76b00-7cc1-11eb-8319-f1df5fddb189.png)

In the Network blade of the Function App there are no access restrictions - 
![image](https://user-images.githubusercontent.com/5489053/109995117-d7be3a80-7cc2-11eb-81e4-c8e3941ec723.png)

I've tried elevating the permissions of the Function App to Owner as well as applying all of the roles to the App.

Also gave the Application "Admin Consent" to default directory.
![image](https://user-images.githubusercontent.com/5489053/109996420-10aadf00-7cc4-11eb-8318-bcb15c992bc3.png)

**Reply by benlabbeus** [reply]
I also tried giving Website Contributor, Web Plan Contributor and Owner permissions to the App and the Function on the Subscription level and am still receiving a 403.

**Reply by shibayan** [reply]
Select the `Login with Azure Active Directory` as the action to perform if the request is not authenticated

**Reply by benlabbeus** [reply]
I'm dumb. You're a legend, ty!
